core/src/upgrade.rs

```diff
@@ -134,6 +135,28 @@ impl<T: AsRef<[u8]>> ProtocolName for T {
     }
 }
 
+/// Wrapper for printing a [`ProtocolName`] that is expected to be mostly ASCII
+///
+/// ```rust
+/// # use libp2p_core::upgrade::DisplayProtocolName;
+/// assert_eq!(DisplayProtocolName("hello").to_string(), "hello");
+/// assert_eq!(DisplayProtocolName("hellö/").to_string(), "hell<C3><B6>/");
+/// ```
+pub struct DisplayProtocolName<N>(pub N);
```

Comment:
I am fine introducing this layer. My preference though would be to fix this the right way, see https://github.com/libp2p/rust-libp2p/issues/2831.

Reply:
Well, “layer” is too big a word — it is a newtype to tweak a trait implementation. Once protocol names are strings, it should go, of course.